prow/config/inrepoconfig.go

```diff
@@ -379,7 +397,16 @@ func (c *InRepoConfigGitCache) ClientFor(org, repo string) (git.RepoClient, erro
 	if cached != nil || err != nil {
 		return cached, err
 	}
-	coreClient, err := c.ClientFactory.ClientFor(org, repo)
+	repoOpts := git.RepoOpts{
+		// We only need ".prow" because the default "cone mode" of sparse
+		// checkouts already include files at the toplevel (which would include
+		// ".prow.yaml").
+		SparseCheckoutDirs: &[]string{".prow"},
```

Comment:
Use inRepoConfigDirName instead of the literal. Also, does the TODO above apply, that you need the top-level file?

Reply:
Oops, will update.

Reply:
Done.